app/models/api_key.rb

```diff
@@ -29,6 +34,18 @@ def mfa_enabled?
     user.mfa_ui_and_api? || mfa
   end
 
+  def rubygem_id=(id)
+    self.ownership = id.blank? ? nil : user.ownerships.find_by!(rubygem_id: id)
```

Comment:
It is possible to handle `ActiveRecord::RecordNotFound` raised here with validation error message?

Reply:
Yes, however when validations are called for save in the api key controller, the current validations will be cleared. So we would need to check for errors beforehand (https://github.com/rubygems/rubygems.org/pull/2944/commits/3f7168f4a527adfafcb500a273d3d36b30e3b349). Let me know if you prefer this rather than rescuing.